### DISCUSSION TITLE
流量识别有更新？ CDN + TLS + WS build 已不再可靠。

### DISCUSSION THREAD

**Discussion by swatchion** [v2fly/v2ray-core#2187] [General]
近期多事之秋。 可以明显感到某墙发威了。

针对域名进行了一系列的识别，当发现有大流量的传输时，会使用 TCP RST packat 重置链接。

大佬们需要针对 CDN + TLS + WS build 做进一步的优化。 能力有限，以上仅是推测。

**Comment by swatchion** [comment]
其他的传输层协议如 grpc 、 mkcp 是否可用？

**Comment by cross-hello** [comment]
@swatchion  Could you share some packets captured via Wireshark? You could remove server info from pcapng file from using the code of https://github.com/cross-hello/Replace-IP-and-Ethc-mac-in-pcapng-file

**Comment by he583899772** [comment]
vmess+tls+ws 已经不可靠，目前封端口，不管是443，还是其他端口，已经到达分钟级别封

**Comment by cross-hello** [comment]
It must be some misconception inside. All our severs survive at the time.

----------------------------------------

*From: *he583899772 ***@***.***>
*To: *v2fly/v2ray-core ***@***.***>
*CC: *Nanyu ***@***.***>; Comment ***@***.***>
*Date: *Nov 30, 2022 13:36:46
*Subject: *Re: [v2fly/v2ray-core] 流量识别有更新？ CDN + TLS + WS build 已不再可靠。 
(Discussion #2187)

**Comment by cross-hello** [comment]
Based one Encryption method charactics, neural network can't effectively 
distinguish.
(see experiment here: 
https://github.com/XTLS/Xray-core/discussions/1295#discussioncomment-4164888
)
So it may be based on packet length combining with time or something 
others. Not matter what it is, absolutely accurate is impossible.
In the high react of blocking it must block other normal service at the 
same time.

----------------------------------------

*From: *he583899772 ***@***.***>
*To: *v2fly/v2ray-core ***@***.***>
*CC: *Nanyu ***@***.***>; Comment ***@***.***>
*Date: *Nov 30, 2022 13:36:46
*Subject: *Re: [v2fly/v2ray-core] 流量识别有更新？ CDN + TLS + WS build 已不再可靠。 
(Discussion #2187)

**Comment by MakeFortuneQuietly** [comment]
我目前人不在国内，谁能帮测下我的vless+tls+ws+cdn+tor节点？是个人闲置欧服

**Reply by jiangangwu** [reply]
我可以

**Reply by MakeFortuneQuietly** [reply]
订阅
https://dushuxyz.ml/hyperlinks/freedom/vl-sub

**Comment by swatchion** [comment]
if you want to keep your line live long, do not use it for bt download.

It works fine before I use it for download. The traffic amount may cause
the trigger happened.

**Comment by MJamshidnejad** [comment]
Something happened to me too. We need to think about ESNI.

**Reply by kingwilliam** [reply]
在14億人大國 已法規明文 絕對禁止 ESNI

**Reply by MJamshidnejad** [reply]
Something similar happened here. Any packet which is set up Host or SNI with filtered domain get dropped. However it doesn't happen in local servers....Just mobile and broadband services!

**Comment by Xyncgas** [comment]
You don't know why, you don't have packet captured, this is a nothing burger though

**Comment by cross-hello** [comment]
Then what about E(ncrypted) C(lient) H(ello)?

----------------------------------------

*From: *kingwilliam ***@***.***>
*To: *v2fly/v2ray-core ***@***.***>
*CC: *Nanyu ***@***.***>; Comment ***@***.***>
*Date: *Dec 7, 2022 00:10:01
*Subject: *Re: [v2fly/v2ray-core] 流量识别有更新？ CDN + TLS + WS build 已不再可靠。 
(Discussion #2187)

**Reply by kingwilliam** [reply]
Date: August 07, 2020

en : https://gfw.report/blog/gfw_esni_blocking/en/
cn : https://gfw.report/blog/gfw_esni_blocking/zh/

ESNI:
```
- We confirm that the Great Firewall (GFW) of China has recently begun blocking ESNI—one of the foundational features of TLS 1.3 and HTTPS.

- 我们确认中国的防火长城（GFW）已经开始封锁ESNI这一TLS1.3和HTTPS的基础特性
```


ECH:
```
- As informed by an anonymous reviewer on the riseup pad, the currently deployed ESNI uses extension value 0xffce (see Section 8.1). However, the newer ECH uses extension value 0xff02, 0xff03 and 0xff04(Section 11.1). We confirm no censorship has been observed on these extension values yet.

- 现有的ESNI使用0xffce作为扩展值（详见 Section 8.1）。 但时新的ECH使用0xff02，0xff03和0xff04作为扩展值（Section 11.1). 我们确认这些新的扩展值还未被封锁。
```

**Reply by MJamshidnejad** [reply]
> Date: August 07, 2020
> 
> en : https://gfw.report/blog/gfw_esni_blocking/en/ cn : https://gfw.report/blog/gfw_esni_blocking/zh/
> 
> ESNI:
> 
> ```
> - We confirm that the Great Firewall (GFW) of China has recently begun blocking ESNI—one of the foundational features of TLS 1.3 and HTTPS.
> 
> - 我们确认中国的防火长城（GFW）已经开始封锁ESNI这一TLS1.3和HTTPS的基础特性
> ```
> 
> ECH:
> 
> ```
> - As informed by an anonymous reviewer on the riseup pad, the currently deployed ESNI uses extension value 0xffce (see Section 8.1). However, the newer ECH uses extension value 0xff02, 0xff03 and 0xff04(Section 11.1). We confirm no censorship has been observed on these extension values yet.
> 
> - 现有的ESNI使用0xffce作为扩展值（详见 Section 8.1）。 但时新的ECH使用0xff02，0xff03和0xff04作为扩展值（Section 11.1). 我们确认这些新的扩展值还未被封锁。
> ```

So we've reached a dead end! right?

**Reply by kingwilliam** [reply]
i don't think so, just can't use tls 1.3 some special feature

**Reply by MJamshidnejad** [reply]
> i don't think so, just can't use tls 1.3 some special feature

Do you have any solutions for the problem that I mentioned above? I was thinking about ESNI, but it doesn't seem like a real solution. 
https://github.com/v2fly/v2ray-core/discussions/2187#discussioncomment-4370358
